src/js/listeners/post/searchListing.js

```diff
@@ -25,6 +29,51 @@ export async function searchListings() {
     else {
       renderNoListings();
     }
- 
   };
 }
+
+export async function filterListings() {
+
+    const arrayOfListings = await getListOfListings();
+
+    function filter(daysInMilliseconds) {
+      const filteredListings = arrayOfListings.filter((listing) => {
+      const dateCreated = new Date(listing.created);
+      const currentDate = new Date();
+      const timeDifference = currentDate.getTime() - dateCreated.getTime();
+      if (timeDifference <= daysInMilliseconds) {
+        return true
+      }
+     });
+     if (filteredListings.length > 0) {
+      renderListings(filteredListings)
+     } else {
+      renderNoListings()
+     }
+    }
+    
+    const btnAll = document.querySelector('#allListings');
+    const btnLatest = document.querySelector('#latestListings');
+    const btnFourteen = document.querySelector('#fourteenDaysListings');
+    const btnThirty = document.querySelector('#thirtyDaysListings');
+
+    const fourDays = 4 * 24 * 60 * 60 * 1000;
+    const fourteenDays = 14 * 24 * 60 * 60 * 1000;
+    const thirtyDays = 30 * 24 * 60 * 60 * 1000;
```

Comment:
This looks a lot better, but there is still some repetition in your code.
Consider declaring 24 * 60 * 60 * 1000 as its own constant to be reused in these three lines